### PR TITLE
Minor updates for academic applications (Compatibility with Pandoc and Papers)

### PR DESCRIPTION
**Pull request by retrography**
GitHub Markdown is used for academic applications. Here I have tried to fix two bothering issues:

1) Pandoc front-matters typically end with '...' rather than the standard '---'. I have added this to the specification. 

See in-line comments.

2) Papers citation software generates Pandoc citation keys in [@Author:year] format. Standard GFM does not recognize ':' as a character that can be used in a Markdown mention. The reason why Papers turned to ':' delimited citekeys is [an interesting read on its own](http://support.mekentosj.com/kb/read-write-cite/universal-citekey). I have added support for this type of citekey so that syntax highlighting works correctly for Papers citekeys in Atom. 

See in-line comments.

### CODE REVIEW

**Comment by kevinsawicki** [comment]
Looks like the tests are failing, mind updating them?

**Comment by retrography** [comment]
Hopefully this fixes the issue. What flavour of Regex does the CSON processor use?

**Comment by kevinsawicki** [comment]
> What flavour of Regex does the CSON processor use?

[Oniguruma](http://www.geocities.jp/kosako3/oniguruma/doc/RE.txt)

File: grammars/gfm.cson [hunk 1]
```diff
@@ -145,7 +145,9 @@
   }
   {
     'begin': '\\A---$'
-    'end': '^---$'
+    # Added '...' as an acceptable end-block
```
Comment (kevinsawicki) [inline grammars/gfm.cson]: Would you mind removing the comments? The `git blame` on this file will trace back to this PR which will tell the story of why this is here.

Comment (retrography) [inline grammars/gfm.cson]: done

File: grammars/gfm.cson [hunk 2]
```diff
@@ -145,7 +145,7 @@
   }
   {
     'begin': '\\A---$'
-    'end': '^---$'
+    'end': '^---|\\.\\.\\.$'
```
Comment (kevinsawicki) [inline grammars/gfm.cson]: Sorry, one last thing, I think this should be `^(---|\\.\\.\\.)$'  so that the lines terminator match both styles.

Comment (retrography) [inline grammars/gfm.cson]: Done. Although it worked without adding a group according to RegexPlanet's Ruby processor. Thanks for the followup.

**Comment by kevinsawicki** [comment]
Thanks! :ship:

**Pull request merged**